### ISSUE TITLE
Validation check in form

### ISSUE THREAD

**Issue by chuenlok** [campsych/concerto-platform#124]
Hi Przmek,

I think there are a few problems with the form validation check. If the `buttons_template` is defined as `buttons_single`, the form seems skipping the validation completely. It would be great if there are some descriptions about what they are and what they will affect.

The second problem is that the validation message is only showing the last fail field. For example, if three fields are not passing the validation, the only fail message shown is the last field. I think the line 
```
alert = concerto.template.join(templateId=form_alert_template,params=list(alert=form[[i]]$alert))
```
should be 
```
alert = paste(alert, concerto.template.join(templateId=form_alert_template,params=list(alert=form[[i]]$alert)), sep="")
```

Also, normally if some fields fail the validations, other fields should hold what they have before submitted. For instance, we have two fields, age(input) and gender(select). If gender is selected, and age is empty, after the form is submitted and failed, the gender field will be unselected again. 

Maybe the line 
```
control = concerto.template.join(templateId=form_field_select_template,params=list(id=id,name=name,options=toJSON(form[[i]]$definition$select_options)))
``` 
should have a parameter called value? I haven't drilled into the code so I am not sure the correct way to fix that.

Thank you very much :)

**Comment by chuenlok** [comment]
Another suggestion to this repository, you may try to have an issue template so that issues are better managed. You can refer to https://help.github.com/articles/creating-an-issue-template-for-your-repository/ and https://help.github.com/articles/creating-an-issue-template-for-your-repository/ 

You can ignore me this annoying person if you don't like this :P

**Comment by przemyslawlis** [comment]
Hi chuenlok,

> I think there are a few problems with the form validation check. If the buttons_template is defined as buttons_single, the form seems skipping the validation completely. It would be great if there are some descriptions about what they are and what they will affect.

^ that's because **form** requires a button named **submit** on the template. **single_button** template doesn't have it.

> The second problem is that the validation message is only showing the last fail field. For example, if three fields are not passing the validation, the only fail message shown is the last field.

^ all alerts are now shown after 4c22483ff71d54b059d5f36283586e402a106d91

> Also, normally if some fields fail the validations, other fields should hold what they have before submitted. For instance, we have two fields, age(input) and gender(select). If gender is selected, and age is empty, after the form is submitted and failed, the gender field will be unselected again.

^ this was fixed in 498e1f0dd37279d8738c132ae3feb2aab7de8a14

Both of the commits above are not yet part of **master** branch.

**Comment by przemyslawlis** [comment]
> Another suggestion to this repository, you may try to have an issue template so that issues are better managed. You can refer to https://help.github.com/articles/creating-an-issue-template-for-your-repository/ and https://help.github.com/articles/creating-an-issue-template-for-your-repository/

That is very helpful. Thanks for that! I'll surely add these to the repo.
